cob_cartesian_controller/src/cartesian_controller.cpp

```diff
@@ -29,430 +29,366 @@
         {
-            ROS_INFO("...service failed");
+            actionAbort(false, "Failed to start traccking");
+            return;
         }
-    }
-
-}
-
-void CartesianController::stopTracking()
-{
-    std_srvs::Empty srv_save_stop;
-    srv_save_stop.request;
-    if(tracking_)
-    {
-        if(stop_tracking_.call(srv_save_stop))
+        
+        // Execute path
+        if(!posePathBroadcaster(cartesian_path))
         {
-            ROS_INFO("... service stopped!");
-            tracking_ = false;
+            actionAbort(false, "Failed to execute path for 'move_lin'");
+            return;
         }
-        else
+        
+        // De-Activate Tracking
+        if(!stopTracking())
         {
-            ROS_ERROR("... service stop failed! FATAL!");
+            actionAbort(false, "Failed to stop traccking");
+            return;
         }
+        
+        actionSuccess(true, "move_lin succeeded!");
     }
-}
-
-void CartesianController::goalCB()
-{
-    tf::TransformListener listener;
-    TrajectoryInterpolator TIP(update_rate_);
-    std::vector <geometry_msgs::Pose> pos_vec;
-    geometry_msgs::Pose actual_tcp_pose;
-    double roll, pitch, yaw;
-    cob_cartesian_controller::CartesianActionStruct action_struct;
-
-    ROS_INFO("Received a new goal");
-
-    if (as_->isNewGoalAvailable())
+    else if(action_struct.name == "move_circ")
     {
-        action_struct = acceptGoal(as_->acceptNewGoal());
-        tracking_goal_ = true;
-
-        if(action_struct.name == "move_lin")
+        ROS_INFO("move_circ");
+        
+        if(!trajectory_interpolator_->circularInterpolation(cartesian_path, action_struct.move_circ))
         {
-            ROS_INFO("move_lin");
-
-            cob_cartesian_controller::MoveLinStruct move_lin = convertMoveLinRelToAbs(action_struct.move_lin);
+            actionAbort(false, "Failed to do interpolation for 'move_circ'");
+            return;
+        }
+        
+        // Publish Preview
+        utils_.previewPath(cartesian_path);
 
-            // Interpolate the path
-            if(TIP.linearInterpolation(pos_vec, move_lin))
-            {
-                //Broadcast the linear path
-                try
-                {
-                    posePathBroadcaster(pos_vec);
-                    actionSuccess();
-                }
-                catch (errorException& e)
-                {
-                    ROS_WARN("%s",e.what());
-                    actionAbort();
-                }
-            }
-            else
-            {
-                ROS_ERROR("Error while interpolation linear path.");
-                tracking_goal_ = false;
-            }
+        // Activate Tracking
+        if(!startTracking())
+        {
+            actionAbort(false, "Failed to start traccking");
+            return;
         }
-        else if(action_struct.name == "move_circ")
+        
+        // Move to start
+        if(!movePTP(cartesian_path.poses.at(0), 0.03))
```

Comment:
as I said...after we switched to using the FrameTracker ActionServer these kind of checks and therefore these kind of parameters will become obsolete here...and in cob_frame_tracker they are already dynamic reconfigurable...